lib/puppetserver/ca/local_certificate_authority.rb

```diff
@@ -66,6 +66,14 @@ def load_ssl_components(loader)
         @cert_bundle = loader.certs
         @key = loader.key
         @cert = loader.cert
+
+        if loader.crl.nil?
+          loader.crl = create_crl_for(@cert, @key)
```

Comment:
`load_ssl_components` is called in `initialize`. I think this creating of the crl is behavior that is desirable in the import action, but not necessarily in every case that we'd instantiate the local CA. I think this logic should be moved to, or at least driven from, the import action in a way that avoids it being called on every `LocalCertificateAuthority.new`.

Reply:
I initially tried to do it that way, but the problem there is that `LocalCertificateAuthority` owns the logic for creating CRLs which depends on the `@digest` that the CA loads from Puppet configuration. The CA also can't create a CRL until the cert and private key have been loaded.

One way to cut the knot may be to move this logic to a different method on `LocalCertificateAuthority`, say `initialize_ssl_components`, that is only called by the `import` action. Would that be a workable solution?

Reply:
sounds good!

Reply:
The CRL creation has been moved to a separate `initialize_ssl_components` action.